src/doc/rustdoc/src/documentation-tests.md

```diff
@@ -233,6 +246,17 @@ not actually pass as a test.
 # fn foo() {}
 ```
 
+`compile_fail` tells `rustdoc` that the compilation should fail. If it
+compiles, then the test will fail. However please note that a code
+failing with the current rustc version may works in a future release.
```

Comment:
Light phrasing nit: "However please note that code failing with the current Rust release may work in a future release, as new features are added."